### PR TITLE
Experimental

### PR DESCRIPTION
**Pull request by devmaxde**
## Type of change

- [ ] Bug fix
- [ ] Feature request
- [ ] Documenting code

## Checklists

- [ ] I referenced the according Issue
  - [ ] I created an Issue
  - [ ] I put a link in the Section [Related Issue](#related-issue)
- [ ] I tested my code with multiple browsers
- [ ] My commit messages are good and understandable

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,9 @@
-# Please fill out with content
\ No newline at end of file
+FROM python:3.8.2-alpine3.10
+COPY . /microservice/
+WORKDIR /microservice/
+RUN pip install --upgrade pip
+
+RUN apk add --update libxml2-dev libxslt-dev gcc g++ bash vim nano
```
Comment (TheCataliasTNT2k) [inline Dockerfile]: Remove vim and nano

File: report/Blueprints/error_handler.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+app_error = Blueprint('error-handler',__name__)
+
+@app_error.app_errorhandler(404)
+def handle_404_error(e):
+    return jsonify({'error':'Page not found'}),404
+
+@app_error.app_errorhandler(405)
+def handle_405_error(e):
+    return jsonify({'error':'Method not allowed'}),404
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/error_handler.py]: change 404 to 405

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+flask
+pyjwt
+flask_sqlalchemy
+flask_marschmallow
+marshmallow-sqlalchemy
+pyjwt
```
Comment (TheCataliasTNT2k) [inline requirements.txt]: use either pipenv or specify version

File: report/__init__.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import os
+import sys
+
+app = Flask(__name__)
+
+try:
+    db_path = "mysql://" + os.environ.get("DATABASE_USERNAME") + ":" + os.environ.get("DATABASE_PASSWORD") + "@" + os.environ.get("DATABASE_HOSTNAME") + ":" + os.environ.get("DATABASE_PORT") + "/" + os.environ.get("DATABASE_DATABASE")
+    app.config['SQLALCHEMY_DATABASE_URI'] = db_path
+except TypeError:
+    sys.exit(0)
```
Comment (TheCataliasTNT2k) [inline report/__init__.py]: 0 means no error

File: report/__init__.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from flask_sqlalchemy import SQLAlchemy
+from flask_marshmallow import Marshmallow
+import os
+import sys
+
+app = Flask(__name__)
+
+try:
+    db_path = "mysql://" + os.environ.get("DATABASE_USERNAME") + ":" + os.environ.get("DATABASE_PASSWORD") + "@" + os.environ.get("DATABASE_HOSTNAME") + ":" + os.environ.get("DATABASE_PORT") + "/" + os.environ.get("DATABASE_DATABASE")
+    app.config['SQLALCHEMY_DATABASE_URI'] = db_path
```
Comment (TheCataliasTNT2k) [inline report/__init__.py:14]: maybe add encoding?
pymysql or mysql needed as driver

File: report/Blueprints/routes.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+
+@main_page.route('/')
+def home():
+    return "hello World"
+
+@main_page.route('/ticket/create', methods=['PUT'])
+def ticked_create():
+    data = request.json
+    if data is not None and all(key in data for key in ('session', 'task_id','title','body','TicketType')):
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py]: flip if and remove print

File: report/Blueprints/routes.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import os
+
+main_page = Blueprint('main-routes', __name__)
+
+
+@main_page.route('/')
+def home():
+    return "hello World"
+
+@main_page.route('/ticket/create', methods=['PUT'])
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py]: change to post request

File: report/Blueprints/routes.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+@main_page.route('/ticket/delete/<id>', methods=['DELETE'])
+def ticked_delete(id):
+    data = request.headers.get('session')
+    try:
+        userdata = jwt.decode(data['session'],os.environ.get('JWT_SECRET') , algorithms=['ES256'])
+    except Exception:
+        return jsonify({'error':'INVALID_SESSION'}),400
+    
+    ticket = Ticket.query.get(id)
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py]: test and surround with try / except

File: report/Blueprints/routes.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    db.session.delete(ticket)
+    db.session.commit()
+    return jsonify({}), 200
+
+@main_page.route('/ticket/list')
+def ticked_list():
+    return jsonify({}), 404
+
+
+@main_page.route('/ticket/edit/<id>')
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py]: put request

File: report/Blueprints/routes.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    if data is not None and all(key in data for key in ('session', 'task_id', 'title', 'body', 'TicketType')):
+        print("Works")
+    else:
+        return jsonify({'error': 'Missing Keys'}), 510
+
+    try:
+        userdata = jwt.decode(data['session'], os.environ.get('JWT_SECRET'), algorithms=['ES256'])
+    except Exception:
+        return jsonify({'error': 'Invalid Session'}), 510
+    ticked_q = Ticket.query.filter_by(ticketId=id).first()
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py]: try for None

File: report/Blueprints/routes.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    try:
+        userdata = jwt.decode(data['session'],os.environ.get('JWT_SECRET') , algorithms=['ES256'])
+    except Exception:
+        return jsonify({'error':'INVALID_SESSION'}),400
+    
+    ticket = Ticket(data['task_id'],data['title'],data['body'],data['TicketType'])
+    db.session.add(ticket)
+    db.session.commit()
+    return jsonify({}), 200
+
```
Comment (TheCataliasTNT2k) [inline report/Blueprints/routes.py:34]: test for needed permission in all routes

**Review by TheCataliasTNT2k — CHANGES_REQUESTED** [review]

File: report/Blueprints/routes.py [hunk 12]
```diff
@@ -1,29 +1,31 @@
...
 
 main_page = Blueprint('main-routes', __name__)
 
 
-@main_page.route('/')
+@main_page.route('/', methods=['POST'])
 def home():
     return "hello World"
 
 @main_page.route('/ticket/create', methods=['POST'])
```
Comment (MarcelCoding) [inline report/Blueprints/routes.py]: Remove `/create`

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Review by MarcelCoding — APPROVED** [review]

**Pull request merged**